makefile

```diff
@@ -1,33 +1,35 @@
 SHELL=/bin/bash
 
 # i.e. fedora28.yaml
-ALL_TEMPLATES=$(wildcard templates/*.yaml)
+ALL_META_TEMPLATES=$(wildcard templates/templates/*.yaml)
+ALL_TEMPLATES=$(wildcard templates/*.dist.yaml)
 ALL_PRESETS=$(wildcard presets/*.yaml)
 SOURCES=$(ALL_TEMPLATES) $(ALL_PRESETS)
+METASOURCES=$(ALL_META_TEMPLATES) $(ALL_PRESETS)
 
 # i.e. fedora28
-ALL_GUESTS=$(ALL_TEMPLATES:templates/%.yaml=%)
+ALL_GUESTS=$(ALL_TEMPLATES:templates/%.dist.yaml=%)
 
 
 TEST_SYNTAX=$(ALL_GUESTS)
 TEST_UNIT=$(ALL_GUESTS)
 ifeq ($(TEST_FUNCTIONAL),ALL)
-TEST_FUNCTIONAL=fedora28 ubuntu1804 opensuse15 rhel75
+TEST_FUNCTIONAL=fedora-generic-small ubuntu-generic-small opensuse-generic-small rhel7-generic-small centos7-generic-small
```

Comment:
Here you should only list guests which can be tested. Which we know which works.